File: api/app/schemas/flight.py
from __future__ import annotations

import uuid
from datetime import date

from pydantic import BaseModel


class SlotResponse(BaseModel):
    slot_id: int
    label: str
    flight_number: str
    gate: str
    flight_id: uuid.UUID | None = None
    aircraft_id: uuid.UUID | None = None
    aircraft_model: str | None = None
    total_seats: int = 0
    available_seats: int = 0


class SlotFlightResponse(BaseModel):
    flight_id: uuid.UUID
    flight_number: str
    gate: str
    departure_date: date
    departure_time: str


class FlightResponse(BaseModel):
    flight_id: uuid.UUID
    flight_number: str
    gate: str
    departure_date: date
    departure_time: str
    aircraft_model: str
    total_seats: int
    available_seats: int
    is_active: bool

File: client/src/types/api/flight.ts
export type SlotResponse = {
  slot_id: number
  label: string
  flight_number: string
  gate: string
  flight_id: string | null
  aircraft_id: string | null
  aircraft_model: string | null
  total_seats: number
  available_seats: number
}

export type SlotFlightResponse = {
  flight_id: string
  flight_number: string
  gate: string
  departure_date: string
  departure_time: string
}

export type FlightResponse = {
  flight_id: string
  flight_number: string
  gate: string
  departure_date: string
  departure_time: string
  aircraft_model: string
  total_seats: number
  available_seats: number
  is_active: boolean
}
